### DISCUSSION TITLE
Connectivity v2 causing error message "Oh no, it looks like you are offline"

### DISCUSSION THREAD

**Discussion by GuiltyRiceHusk** [community/community#124751] [Codespaces]
### Select Topic Area

Product Feedback

### Body

I have trouble accessing CS50's codespace, with the error message being "Oh no, it looks like you are offline", error details being "Error: Network Error".

A few days ago when I turned the Connectivity v2 feature off, I was able to access it. But now since it is enabled by default for all GitHub users and can't be disabled, it is showing this error message again. 

My attempts to solve but in vain: 
1. do not work on safari/firefox. The latter shows "Oh no, something went wrong!".
2. do not work with vpn, or without one, or with another vpn. (I'm in China so I have to use a vpn to access everything)
3. I did not turn on my firewall(I'm using mac).

This is how the 'console' in the Network tab of the Developer tools in the browser looks like.
<img width="927" alt="Screen Shot 2024-05-20 at 16 15 08" src="https://github.com/community/community/assets/138004977/b9b1a5e1-c67d-40e2-8a1c-215c4ec888a5">
<img width="927" alt="Screen Shot 2024-05-20 at 16 15 27" src="https://github.com/community/community/assets/138004977/d5f4e703-c3b0-41b4-a404-21909be1f02c">

CS50 has a [troubleshooting](https://cs50.readthedocs.io/cs50.dev/#i-m-seeing-oh-no-it-looks-like-you-are-offline-error-message) for this specific situation but I am new in coding and don't really know how to check whether certain domains are blocked as it says. 

I see my problem is pretty similar to #112351, but this one hasn't been solved yet. Please let me know if more information is needed to solve the problem. Thank you very much.

**Comment by slugsack** [comment]
Same thing happening for me. Tried different browser, different device, updating browser, turning off VPN, changing VPN location, clearing cache and cookies, restarting codespace, restarting computer, disabling enhanced tracking protections in Firefox, disabling my adblocker, and loading in desktop VSC. Only turning v2 off worked, when it was still an option.

The only success I've had beyond that is accessing the codespace via my phone's data, but with Canadian data rates, this is not viable longer term than about 5 minutes.

**Comment by GuiltyRiceHusk** [comment]
God, hope you find the solution soon... Sorry for the late reply, but please consider using another VPN! I did this a few days ago and it worked!

**Comment by slugsack** [comment]
Thank you for the idea - do you happen to use Mullvad, or another VPN with DNS blocking?

I had tried turning my VPN off already, but this time I tried closing the program entirely. The space then loaded fine. I dug around in the settings to see if there might be some setting that the VPN program uses even when "off", and I suspect it may be the DNS blocker for tracking. This might make sense given that VSC has telemetry and the blocklists Mullvad uses include a list for Microsoft tracking specifically. This doesn't explain why it worked when I was using mobile data, though. I still had my VPN on then.

This isn't an acceptable fix, in any case. If Connectivity V2 demands use of something Mullvad is blocking for privacy, that makes me uneasy.

**Comment by WilmarkJ** [comment]
I was getting the same error, and i realized it was my adblocker in Chrome causing it. After i white listed it it went away.

**Reply by hurleyp1** [reply]
Thanks, this was my problem. I had to add my particular codespace url as an exception to firefox's strict Browser Privacy > Enhanced Tracking Protection

**Reply by MyJourneyToTheStars666** [reply]
Worked for me as well (added site to exceptions)

**Reply by bwendell** [reply]
Thanks for mentioning this! Changing my Firefox "Enhanced Privacy Tracking" to Standard from Strict also worked.

**Reply by Nav-R-1** [reply]
Adding the site to exceptions for enhanced tracking protection worked for me as well.

**Comment by R3posit** [comment]
Hmm why?

**Reply by WilmarkJ** [reply]
I guess my adblocker was blocking important scripts on the codespaces page that it assumed was malicious.

**Comment by Alm0stEthical** [comment]
Check Content Security Policies: If you're not familiar with checking the domains blocked as per the CS50 troubleshooting guide, you can use the browser's developer tools to inspect the network activity. In the "Network" tab, look for failed requests (red lines), which might indicate blocked domains.

Try Different VPN Servers: Since you're accessing from China, using a different VPN server may help bypass any geo-blocking issues. Additionally, certain VPNs might be more reliable than others for accessing GitHub or CS50 resources.

Consult CS50 Troubleshooting: Since there is a CS50 troubleshooting guide for your issue, it may help to follow the steps there closely, especially regarding checking if certain domains are blocked by the browser or VPN.

**Comment by Frulfump** [comment]
Thanks to those that suggested adding exceptions for Firefox Enhanced Tracking Protection
Annoying having to add each new code space to Firefox's exceptions for Enhanced Tracking Protection since it's a new URL for each new Codespace. Edit: seems like you can add just the `github.dev` address as an exception.

I think the culprit is `https://euw.rel.tunnels.api.visualstudio.com` (the part before .rel is presumably region dependant) and on the endpoint `https://euw.rel.tunnels.api.visualstudio.com/tunnels/<codespace name>?api-version=2023-09-27-preview`

The request fails with this in the console `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://euw.rel.tunnels.api.visualstudio.com/tunnels/<codespace name>/ports/16635?api-version=2023-09-27-preview. (Reason: CORS request did not succeed). Status code: (null)`

and the reason in the network tab is `Tracking` not sure if it's actually a tracker and Firefox needs to change something or if GitHub can do something on their side?

Opened a web compat issue here https://github.com/webcompat/web-bugs/issues/160595 but should maybe be a BugZilla issue instead, let's see if something comes of it first.

**Comment by rwonkensv2** [comment]
Anyone still having the same issue, what worked for me was disabling adblocking extensions and disabling enhanced tracking protection (on firefox)

**Comment by fn5** [comment]
I had to turn off Strict Enhanced Tracking Protection in Edge for my Codespace to work today.
